File: Lab1/Lab1/Result.cs
using System.Text;

namespace Lab1;

public class Result
{
    private int totalValue;
    private int totalWeight;
    private List<Item> result;

    public Result()
    {
        this.totalValue = 0;
        this.totalWeight = 0;
        result = new List<Item>();
    }

    public void insertIntoList(Item i)
    {
        result.Add(i);
        totalValue += i.Value;
        totalWeight += i.Weight;
    }

    public int getTotalValue()
    {
        return totalValue;
    }

    public int getTotalWeight()
    {
        return totalWeight;
    }

    public List<Item> getResultList()
    {
        return result;
    }

    public string printResult()
    {
        StringBuilder text = new StringBuilder();

        Console.WriteLine("===========Result===========");

        text.AppendLine("Total value: " + totalValue);
        text.AppendLine("Total weight: " + totalWeight);
        text.AppendLine("======Selected items======");

        foreach (var item in result)
        {
            Console.WriteLine(item + " | ratio: "+item.Ratio);
            text.AppendLine(item + " | ratio: " + item.Ratio);
        }
        Console.WriteLine("Total value: {0} | Total weight: {1}",totalValue,totalWeight);

        return text.ToString();
    }
}

File: src/main/java/org/example/knapsack/Result.java
package org.example.knapsack;

import java.util.ArrayList;
import java.util.List;

public class Result {
    protected List<Item> items;
    protected int amount;
    protected int totalValue;
    protected int totalWeight;


    public Result(){
        items = new ArrayList<>();
        amount=0;
        totalValue=0;
        totalWeight=0;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("Result: \n");
        sb.append("Amount: ").append(amount).append("\n");
        sb.append("Total value: ").append(totalValue).append("\n");
        sb.append("Total weight: ").append(totalWeight).append("\n");
        sb.append("Selected items: \n");
        for(Item item : items) {
            sb.append(item).append("\n");
        }
        return sb.toString();
    }
}
